aiida/cmdline/commands/cmd_verdi.py

```diff
@@ -82,9 +82,26 @@ def get_command(self, ctx, cmd_name):
         return None
 
 
+def version_callback(ctx, param, value):  # pylint: disable=unused-argument
+    """Callback for supplying version information"""
+    if not value or ctx.resilient_parsing:
+        return
+    from aiida import __version__
+    click.echo(f'AiiDA version {__version__}')
+    ctx.exit()
+
+
 @click.command(cls=MostSimilarCommandGroup, context_settings={'help_option_names': ['-h', '--help']})
 @options.PROFILE(type=types.ProfileParamType(load_profile=True))
-@click.version_option(None, '-v', '--version', message='AiiDA version %(version)s')
```

Comment:
I think there's a simpler way to do this by just replacing `None` with `aiida.__version__` in the `version_option`.

Reply:
This is more customisable

Reply:
Guess the question the is whether we need that customizability.

It's ok for me to go either way, but we should maybe add a comment somewhere pointing to the problem with a bare `version_option`. I can see someone a few releases from now thinking "hey, I can just replace this with `click.version_option`.

Reply:
Agree with @greschd , we should at least add a comment why `click's` built in does not work if we go the custom route. That being said, I also prefer the more simple implementation. Don't see why preparing for a potential future need for customizability is necessary right now. If it can be solved as suggested by just replacing `None` with `aiida.__version__` then I would go that route

Reply:
so be it, changed

Reply:
I think a short comment regarding the click issue still couldn't hurt, something like "We need to pass the `__version__` explicitly because click's default can be wrong when the project is installed in editable mode, see issue #4556."

Reply:
ok fair, this was all clearer with the initial (explicit) implementation I wrote, rather than calling this "faulty" black box convenience function 😛

Reply:
I think it is fine, there is a reason the `click>version_option` provides an argument to not automatically determine it but specify explicitly the version. I agree that a comment is still useful, because the reason _why_ we are passing in an explicit version may not be directly obvious

Reply:
hows that 4bfac3a

Reply:
Beautiful, thanks @chrisjsewell